src/librustc_borrowck/borrowck/check_loans.rs

```diff
@@ -745,6 +745,26 @@ impl<'a, 'tcx> CheckLoanCtxt<'a, 'tcx> {
                                                      use_kind, lp_base);
             }
             LpExtend(ref lp_base, _, LpInterior(InteriorField(_))) => {
+                match lp_base.to_type().sty {
+                    ty::ty_struct(def_id, _) | ty::ty_enum(def_id, _) => {
+                        if ty::has_dtor(self.tcx(), def_id) {
+                            // In the case where the owner implements drop, then
+                            // the path must be initialized to prevent a case of
+                            // partial reinitialization
```

Comment:
Off the top of my head, it was possible but the error was a more generic 'use of moved value `foo.bar`'.

I reasoned that the improved error message from the original changeset might be preferable but I'm willing to change it for simpler code.